src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (item === null || item === undefined) {
+        item = '';
+      }
+
+      if (i < this.length - 1) {
+        result += `${item}${separator}`;
```

Comment:
You are appending the separator even when the next item may be undefined or null, which will result in trailing separators in the output string. You should ensure that the next item is not null or undefined before appending the separator.